livekit-agents/livekit/agents/utils/codecs/decoder.py

```diff
@@ -148,18 +145,14 @@ def __init__(
         self._input_buf = StreamBuffer()
         self._loop = asyncio.get_event_loop()
 
-        if self.__class__._executor is None:
-            # each decoder instance will submit jobs to the shared pool
-            self.__class__._executor = ThreadPoolExecutor(
-                max_workers=self.__class__._max_workers, thread_name_prefix="AudioDecoder"
-            )
+        self._executor = ThreadPoolExecutor(max_workers=1, thread_name_prefix="AudioDecoder")
```

Comment:
replacing with instance-specific decoder thread so we can shut it down upon aclose